pyprof/nvtx/nvmarker.py

```diff
@@ -252,20 +248,21 @@ def wrapper_func(*args, **kwargs):
                 # Keeps call_id correct when there are nested
                 # monkey patch functions
                 saved_call_id = dlprof.call_id
-                if dlprof.call_id != patch_list[0]:
-                    saved_call_id = patch_list[0]
+                if dlprof.call_id != dlprof.patch_list[0]:
+                    saved_call_id = dlprof.patch_list[0]
                 dlprof.capture_outputs(saved_call_id, result)
                 # Store the callid -> op_name mapping
                 if traceMarker_str is not "":
                     traceMarker_str = traceMarker_str.replace("\'", "\"")
                     traceMarker_dict = json.loads(traceMarker_str)
                     dlprof.call_id_to_op_map[saved_call_id] = traceMarker_dict['funcStack']
-                dprint("Fn {} Incrementing call_id from {} to {}".format(fn_name, dlprof.call_id, dlprof.call_id + 1))
+                dprint(f"Fn {fn_name} Incrementing call_id from {dlprof.call_id} to"\
+                        f" {dlprof.call_id + 1}")
 
-                starting_call_id = patch_list[0]
-                last_call_id     = patch_list.pop()
-                dprint("Ending:callid: {} orig callid {} last_callid {} "\
-                        .format(dlprof.call_id, starting_call_id, last_call_id))
+                starting_call_id = dlprof.patch_list[0]
+                last_call_id     = dlprof.patch_list.pop()
+                dprint(f"Ending:callid: {dlprof.call_id} orig callid {starting_call_id} last_callid"\
+                        f" {last_call_id}")
```

Comment:
Same as above. Move to a helper function and optimize code so that when debug is False, this function is almost a noop.